src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -23,6 +23,180 @@ namespace System
+
+                case StringComparison.CurrentCultureIgnoreCase:
+                    return CultureInfo.CurrentCulture.CompareInfo.Compare(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.InvariantCulture:
+                    return CompareInfo.Invariant.Compare(span, value, CompareOptions.None);
+
+                case StringComparison.InvariantCultureIgnoreCase:
+                    return CompareInfo.Invariant.Compare(span, value, CompareOptions.IgnoreCase);
+
+                case StringComparison.Ordinal:
+                    if (span.Length == 0 || value.Length == 0)
+                        return span.Length - value.Length;
+                    return CompareOrdinalHelper(span, value);
+
+                case StringComparison.OrdinalIgnoreCase:
+                    return CompareInfo.CompareOrdinalIgnoreCase(span, value);
+
+                default:
+                    throw new ArgumentException(SR.NotSupported_StringComparison, nameof(comparisonType));
+            }
+        }
+
+        public static unsafe int IndexOf(this ReadOnlySpan<char> span, ReadOnlySpan<char> value, StringComparison comparisonType, out int matchedLength)
+        {
+            if (value.Length == 0)
+            {
+                StringSpanHelpers.CheckStringComparison(comparisonType);
+                matchedLength = 0;
+                return 0;
+            }
+
+            if (span.Length == 0)
+            {
+                StringSpanHelpers.CheckStringComparison(comparisonType);
+                matchedLength = 0;
+                return -1;
+            }
+
+            int defaultMatchLength = value.Length;
+
+            switch (comparisonType)
+            {
+                case StringComparison.CurrentCulture:
+                    int index = IndexOfCultureHelper(span, value, &defaultMatchLength);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.CurrentCultureIgnoreCase:
+                    index = IndexOfCultureIgnoreCaseHelper(span, value, &defaultMatchLength);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.InvariantCulture:
+                    index = IndexOfCultureHelper(span, value, &defaultMatchLength, invariantCulture: true);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.InvariantCultureIgnoreCase:
+                    index = IndexOfCultureIgnoreCaseHelper(span, value, &defaultMatchLength, invariantCulture: true);
+                    matchedLength = defaultMatchLength;
+                    return index;
+
+                case StringComparison.Ordinal:
+                    matchedLength = defaultMatchLength;
+                    return IndexOfOrdinalHelper(span, value, false);
```

Comment:
Nit: It'd be nice to specify the name of the bool parameter. Same for a few other calls in this file below.